ocaml/xapi/rbac_audit.ml

```diff
@@ -127,33 +108,21 @@ let get_sexpr_arg name name_of_ref uuid_of_ref ref_value : SExpr.t =
 (* used on the master to map missing reference names from slaves *)
 let get_obj_names_of_refs (obj_ref_list : SExpr.t list) : SExpr.t list =
   List.map
-    (fun obj_ref ->
-      match obj_ref with
-      | SExpr.Node
-          [
-            SExpr.String name
-          ; SExpr.String ""
-          ; SExpr.String ""
-          ; SExpr.String ref_value
-          ] ->
-          get_sexpr_arg name
-            ( match get_obj_name_of_ref ref_value with
-            | None ->
-                "" (* ref_value is not a ref! *)
-            | Some obj_name ->
-                obj_name (* the missing name *)
-            )
-            ( match get_obj_uuid_of_ref ref_value with
+    (function
+      | SExpr.(Node [String name; String ""; String ""; String ref_value]) ->
+          let obj_name, uuid =
+            match Ref_index.lookup ref_value with
             | None ->
-                "" (* ref_value is not a ref! *)
-            | Some obj_uuid ->
-                obj_uuid (* the missing uuid *)
-            )
-            ref_value
-      | _ ->
-          obj_ref
-      (* do nothing if not a triplet *)
-    )
+                ("", "")
+            | Some {name_label= None; uuid; _} ->
+                ("", uuid)
+            | Some {name_label= Some name; uuid; _} ->
+                (name, uuid)
```

Comment:
You can't have a obj_name without a uuid (as the old code suggests)?

Reply:
Seems not indeed, looking at Ref_index. The old code unnecessarily turns the uuid into an option.

Reply:
The issue is having a uuid without an object name